src/content/docs/essentials/database.mdx

```diff
@@ -0,0 +1,205 @@
+---
+title: Database
+description: Database and Drift
+---
+
+In almost any project we deal with data, often in substantial quantities. In such cases, it is important
+to decide on the tools that will be used. We need that the data is reliably stored and that we can access
+it without dancing around the fire with tambourines. When working with a cross-platform, it is also important
+to consider that the user can access data on different devices. And in all of this, the lightweight fighter
+[SQLite](#why-sqlite) will come to our aid, who will be "guide" on the right path by
+[Drift library](#what-is-drift).
+
+## Why SQLite?
+
+SQLite is selected for data processing due:
+
+- **Lightweight:** It is a compact library that is easy to deploy and does not require any external dependencies
+  or configuration.
+
+- **Reliability:** Highly reliable, as it is extensively tested and used in production by a large number of
+  applications.
+
+- **Сross-platform:** Compatibility across diverse devices makes it versatile for deployment on various
+  devices, which is important when working with a multi-platform framework such as
+  Flutter because it allows seamless integration between different platforms.
+
+- **Data Analysis:** Efficient for analyzing large datasets using SQL commands, making it a choice for
+  bioinformatics research.
+
+- **Cache for Enterprise Data:** Acts as a cache for enterprise content, reducing latency and network load.
+
+- **Data Transfer Format:** Serves as a compact, cross-platform container for transferring content between
+  systems.
+
+More about SQLite can be found [here](https://www.sqlite.org/whentouse.html).
+
+## What is Drift?
+
+Drift is a robust database library designed for Dart and Flutter applications. It empowers developers with
+advanced features such as type-safe SQL queries, database verification, and seamless migrations. Drift
+leverages a builder and command-line tooling that operates during compile-time, ensuring efficient and
+error-resistant database interactions in Dart and Flutter projects.
+
+### Defining SQL tables
+
+To define a table, we need to create a class that extends Table. The class name will be the name of the
+table. The fields of the class will be the columns of the table. The type of the field will be the type
+of the column.
+
+```dart
+class Spendings extends Table {
+    IntColumn get id => integer().autoIncrement()();
+    RealColumn get price => real().withDefault(const Constant(0.0))();
+    TextColumn get title => text()();
+    TextColumn get currency => text().nullable()();
+    DateTimeColumn get date => dateTime()();
+    BlobColumn get receiptImage => blob().nullable()();
```

Comment:
It is considered a bad practise to store images in databases as BLOB. It is preferred to store links in fs.